### PR TITLE
LPD-20140 Allow changes to be moved from an out of date publication

### PR DESCRIPTION
**Pull request by gislayne-vitorino**
This is an update for: [LPD-20140](https://liferay.atlassian.net/browse/LPD-20140)

Now out of date publications can have its changes moved to another publication
image

### CODE REVIEW

**Comment by gislayne-vitorino** [comment]
ci:test:publications

File: modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/display/context/ViewChangesDisplayContext.java [hunk 1]
```diff
@@ -217,6 +217,27 @@ public List<FDSActionDropdownItem> getFDSActionDropdownItems() {
 				_language.get(_httpServletRequest, "review-change"), "get",
 				"get", null));
 
+		if (_ctCollection.getStatus() == WorkflowConstants.STATUS_EXPIRED) {
```
Comment (david-truong) [inline modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/display/context/ViewChangesDisplayContext.java:220]: Can you rewrite the if check to also include drafts or expired so we don't have the move changes logic twice

**Comment by gislayne-vitorino** [comment]
ci:test:sf

**Comment by gislayne-vitorino** [comment]
ci:test:sf

File: modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/display/context/ViewChangesDisplayContext.java [hunk 2]
```diff
@@ -217,28 +217,15 @@ public List<FDSActionDropdownItem> getFDSActionDropdownItems() {
...
-					).buildString(),
-					"move-folder", "move-changes",
-					_language.get(_httpServletRequest, "move-changes"), "post",
-					"move-changes", null));
-		}
+		boolean statusDraft = false;
 
 		if (_ctCollection.getStatus() == WorkflowConstants.STATUS_DRAFT) {
+			statusDraft = true;
+		}
```
Comment (gislayne-vitorino) [inline modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/display/context/ViewChangesDisplayContext.java:224]: Tried to simplify by just adding the content directly to the boolean 
`boolean statusDraft =  _ctCollection.getStatus() == WorkflowConstants.STATUS_DRAFT `, but SourceFormat didn't allow it and changed to code to this current version

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties [hunk 3]
```diff
@@ -19383,7 +19383,7 @@ this-publication-contains-conflicting-changes-that-must-be-manually-resolved-bef
 this-publication-contains-unapproved-changes=This publication contains unapproved changes.
 this-publication-contains-unapproved-changes-that-must-be-approved-before-publishing=This publication contains unapproved changes that must be approved before publishing.
 this-publication-no-longer-exists=This publication no longer exists.
-this-publication-was-created-on-a-previous-liferay-version.-you-cannot-publish,-revert,-or-make-additional-changes=This publication was created on a previous Liferay version. You cannot publish, revert, or make additional changes. You can only view or delete it.
+this-publication-was-created-on-a-previous-liferay-version.-you-cannot-publish,-revert,-or-make-additional-changes=This publication was created on a previous Liferay version. You cannot publish, revert, or make additional changes. You can only view, delete it or move its changes to another publication.
```
Comment (pei-jung) [inline modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:19386]: The last sentence should be `You can only view, delete, or move changes to another publication.`

File: modules/apps/change-tracking/change-tracking-test/src/testIntegration/java/com/liferay/change/tracking/service/test/CTCollectionLocalServiceTest.java [hunk 4]
```diff
@@ -595,7 +649,16 @@ public void testUndoCTCollection() throws Exception {
 	@DeleteAfterTestRun
 	private CTCollection _ctCollection4;
 
+	@Inject
+	private CTCollectionService _ctCollectionService;
+
+	@Inject
+	private DLFileEntryLocalService _dlFileEntryLocalService;
```
Comment (pei-jung) [inline modules/apps/change-tracking/change-tracking-test/src/testIntegration/java/com/liferay/change/tracking/service/test/CTCollectionLocalServiceTest.java:656]: I don't see usages for this service. Is `DLFileEntryLocalService` needed?

File: modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/display/context/ViewChangesDisplayContext.java (hunk 2 above)
Comment (pei-jung) [reply modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/display/context/ViewChangesDisplayContext.java:224]: Can you try to do it without creating a variable? Simply do 
```
if ((_ctCollection.getStatus() == WorkflowConstants.STATUS_DRAFT) ||
	(_ctCollection.getStatus() == WorkflowConstants.STATUS_EXPIRED))
```

**Review by pei-jung — CHANGES_REQUESTED** [review]

**Comment by gislayne-vitorino** [comment]
ci:test:sf

**Comment by gislayne-vitorino** [comment]
ci:test:publications

**Comment by pei-jung** [comment]
ci:report:2556031852

**Comment by gislayne-vitorino** [comment]
ci:test:sf

**Comment by gislayne-vitorino** [comment]
ci:test:sf

**Review by pei-jung — APPROVED** [review]

**Comment by pei-jung** [comment]
ci:test:relevant

**Comment by pei-jung** [comment]
ci:test:relevant

**Comment by pei-jung** [comment]
ci:test:relevant

**Comment by pyoo47** [comment]
ci:test:relevant

**Comment by pei-jung** [comment]
ci:forward:force

**Pull request closed without merging**